Add extensibility team page (#2601)
sourcegraph/about: handbook/engineering/extensibility/index.md

# Extensibility team

The extensibility team owns our code host and third-party integrations (including our browser extension) and our [Sourcegraph extensions](https://docs.sourcegraph.com/extensions).

## Contact

- [#extensibility-chat](https://app.slack.com/client/T02FSM7DL/C01LZKLRF0C) channel or @extensibility on Slack
- On GitHub, mention or assign issues to the [@sourcegraph/extensibility](https://github.com/orgs/sourcegraph/teams/extensibility) team.

## Ownership

This team is responsible for:

> 1. Bringing the value of Sourcegraph to other developer tools

Reviewer: ```suggestion
1. Bringing the value of Sourcegraph into everywhere you work with code
```
I almost wonder if this is broad enough? Would you call "Stack overflow" or "Stripe API Docs" a developer "tool"? Is Slack a "developer tool" rather than a "workplace tool"? What if we want the extension to recognize code on those sites or in those apps? Just a provocation here.  

"Bringing the value of Sourcegraph with you to everywhere you work with code" is perhaps a little too vague (thoughts?), but I do think it's important that we expand everyone's concept of where Sourcegraph can be useful if we want to truly achieve our broad mission. Alt version: "Bringing the value of Sourcegraph into other tools and websites" but that sounds less exciting

> 1. Bringing the value of Sourcegraph to other developer tools

Reviewer: ```suggestion
1. Bringing the value of Sourcegraph to everywhere you work with code
```
What about if we appear on Stripe API Docs or Stack Overflow? Are docsites "tools"? (Tools about the tools they document?) Is Slack -- for a Slack integration -- a developer tool? I'd prefer to make this much broader in order to encourage people to think of new uses/not limit to traditional developer tools as we see fit! A less-exciting but less-vague version of this is "to other tools and websites"

Author: Good point, I'll limit our use of "developer tools" in the team page, since it can be seen as both vague and narrow

> 1. Bringing the value of other developer tools into Sourcegraph

Reviewer 2: Do these goals intentionally exclude any work on extensions that _don't_ integrate with some external developer tool, like your idea of the code tour extension @tjkandala? I personally think the goals should include/allow extensions like that if we want extensions to become a hit with customers in 2021

Reviewer: [Same thought](https://github.com/sourcegraph/about/pull/2601#discussion_r580711709) :)

The team's areas of responsibility include:

- Sourcegraph extensions
  - Providing a Sourcegraph extension API that enables developers to bring data from their favorite developer tools into their Sourcegraph workflow.

>   - Building useful Sourcegraph extensions on top of the Sourcegraph extension API (for example: Codecov, Datadog, Sentry, Lightstep).

Reviewer: ```suggestion
  - Building and maintaining useful Sourcegraph extensions on top of the Sourcegraph extension API ([Sourcegraph maintained extensions](https://docs.sourcegraph.com/dev/background-information/sourcegraph_extensions)).
```
Possibly this will be a section in the extensions registry soon-ish!

>   - Building useful Sourcegraph extensions on top of the Sourcegraph extension API (for example: Codecov, Datadog, Sentry, Lightstep).

Reviewer: ```suggestion
  - Building and maintaining useful Sourcegraph extensions on top of the Sourcegraph extension API ([Sourcegraph-maintained extensions](https://docs.sourcegraph.com/dev/background-information/sourcegraph_extensions)).
```

  - Documentation and tutorials that enable third party developers to create extensions that are actively used.
  - Building and maintaining the extensions registry and discovery paths to create a community around Sourcegraph extensions.

> - Integrations

Reviewer: ```suggestion
- Code Host Integrations
```
To avoid confusion with https://docs.sourcegraph.com/integration

> - Integrations

Reviewer: ```suggestion
- Code host integrations
```
To avoid confusion with https://docs.sourcegraph.com/integration

>   - Surface code intelligence (and other Sourcegraph data) in code hosts through user installed browser extensions.

Reviewer: ```suggestion
  - Surface code intelligence (and other Sourcegraph data) in code hosts through user-installed browser extensions.
```

  - Add native support for Sourcegraph in code hosts (for example: GitLab) or popular sites where developers look at code examples (e.g. https://reactjs.org, https://pkg.go.dev/) so users don't have to install our browser extension to get code intelligence.

> 

Reviewer: What about the third category, of IDE extensions? (IDE --> Sourcegraph)

> 

Reviewer: What about the third category (currently) of IDE --> Sourcegraph extensions? Is that an IDE integration? (If we want the two categories of "in" vs "out" for extensibility?) In which case, should it be a subsection of the above?

> [List of Sourcegraph-maintained extension repositories](https://docs.sourcegraph.com/dev/background-information/sourcegraph_extensions)

Reviewer: I think you can remove if you link above

> [List of Sourcegraph-maintained extension repositories](https://docs.sourcegraph.com/dev/background-information/sourcegraph_extensions)

Reviewer: I would probably just link this above and delete it here

- [Web workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API)

## Processes

For now, we participate in [Web team processes](../web/index#processes) with the following additions:

### Pair programming

We use pair programming extensively. We're huge believers in pair programming in remote work contexts, so we aim to pair for 15 hours a week.

- TODO

### Weekly Sync

The extensibility team holds weekly syncs.

The meeting notes of extensibility team syncs can be found [in this doc](https://docs.google.com/document/d/1apinxDIp2PdyDHjkr_nBuD7ykfzItVuTvWejiA66sjY/edit?ts=5fb7fd29#).

Before web team syncs, teammates and stakeholders should write down under "Agenda" in the meeting notes document anything that they'd like to bring up for discussion with the whole team.

## Members

- [Joel Kwartler](../../../company/team/index.md#joel-kwartler-he-him) ([Product Manager](../../product/roles/index.md#product-manager)) is the acting PM for this team until we can hire a dedicated PM for it ([apply here](https://jobs.lever.co/sourcegraph/254299f5-f91b-43e2-aa1a-3732963dd296)). Once that happens, Joel will focus entirely on code insights.

> - FQ3 [Engineering Manager](../roles.md#engineering-manager), Jean acting manager until then.

Reviewer 2: Jean is not linked to team page
